File: go-producer/internal/usecases/sendnotification/dtos.go
package sendnotification

import (
	"time"

	"github.com/medeiros-dev/notification-service/go-producer/internal/domain"
)

// SendNotificationInput represents the data needed to send a notification.
type SendNotificationInputDTO struct {
	ID          string             `json:"id"`
	Content     string             `json:"content"`
	Channels    []Channel          `json:"channels"`
	UserProfile domain.UserProfile `json:"user_profile"`
	CreatedAt   time.Time          `json:"created_at"`
}

type Channel struct {
	Type        string `json:"type"`
	Destination string `json:"destination"`
	Enabled     bool   `json:"enabled"`
}

type SendNotificationOutputDTO struct {
	Messages []MessageResponse `json:"messages"`
}

type MessageResponse struct {
	ID          string `json:"id"`
	ChannelType string `json:"channel_type"`
	Status      string `json:"status"`
	Error       string `json:"error"`
}

File: py-producer/app/usecases/dtos.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional
from app.domain.models import UserProfile

class Channel(BaseModel):
    type: str
    destination: str
    enabled: bool = True

class SendNotificationInputDTO(BaseModel):
    id: str
    content: str
    channels: List[Channel]
    user_profile: UserProfile
    created_at: Optional[datetime] = None

class MessageResponse(BaseModel):
    id: str
    channel_type: str
    status: str
    error: Optional[str] = None

class SendNotificationOutputDTO(BaseModel):
    messages: List[MessageResponse] 